lib/matplotlib/image.py

```diff
@@ -421,12 +421,12 @@ def _make_image(self, A, in_bbox, out_bbox, clip_bbox, magnification=1.0,
                 if inp_dtype.kind == 'f':
                     scaled_dtype = A.dtype
                     # Cast to float64
-                    if A.dtype not in (np.float32, np.float16):
-                        if A.dtype != np.float64:
-                            cbook._warn_external(
-                                f"Casting input data from '{A.dtype}' to "
-                                f"'float64' for imshow")
-                        scaled_dtype = np.float64
+                    if A.dtype not in (np.float64, np.float32, np.float16):
+                        # warn if decreasing float precision
```

Comment:
If we are casting to higher bit depth isn't it increasing float precision?

Reply:
I could have made that better, but the only casting the warning really covers is float128 -> float64. Everything else is cast automatically. From what I could tell they were always casting float128 down and ints into floats. float16 through float64 were preserved and passed along as is so I'm wondering if it isn't out of scope here to just cast float16 up.